common/app/templates/inlineJS/nonBlocking/detectAdblock.scala.js

```diff
@@ -0,0 +1,24 @@
+try {
+    ((document, window) => {
+        var ad = document.createElement('div');
+        ad.style.position = 'absolute';
+        ad.style.left = '-9999px';
+        ad.style.height = '10px';
+        ad.innerHTML = '&nbsp;';
+        ad.setAttribute('class', 'ad_unit');
+
+        window.requestAnimationFrame(() => {
+            document.body.appendChild(ad);
+
+            window.requestAnimationFrame(() => {
```

Comment:
Do ad blockers change the value of the `style` attribute? If yes, then you certainly don't need/want to wait until the browser has calculated layout: you can be notified long before, using mutation observers (even mutation events for older browsers—in this case it's rather safe). Then you read the value of the `style` attribute, **not** `getComputedStyle`. Here's a quick [secret gist](https://gist.github.com/regiskuckaertz/2c69fd7df2394c26ea57).